src/main/java/org/folio/services/title/TitleService.java

```diff
@@ -77,4 +87,54 @@ public Future<Tx<PoLine>> updateTitle(Tx<PoLine> poLineTx, String instanceId, DB
         }
       });
   }
+
+  public Future<String> saveTitle(Title title, Conn conn) {
+    if (StringUtils.isBlank(title.getId())) {
+      title.setId(UUID.randomUUID().toString());
+    }
+
+    return conn.getById(PO_LINE_TABLE, title.getPoLineId(), PoLine.class)
+      .compose(poLine -> {
+        log.debug("saveTitle:: A poLine with an id={} was found", poLine.getId());
+        return populateTitle(title, poLine, conn)
+          .compose(aVoid -> conn.getById(PURCHASE_ORDER_TABLE, poLine.getPurchaseOrderId(), PurchaseOrder.class));
+      })
+      .compose(purchaseOrder -> {
+        log.debug("saveTitle:: A purchaseOrder with an id={} was found", purchaseOrder.getId());
+        title.withAcqUnitIds(purchaseOrder.getAcqUnitIds());
+
+        log.debug("saveTitle:: Creating new title record with id={}", title.getId());
+        return conn.save(TITLES_TABLE, title.getId(), title)
+          .onSuccess(rowSet -> log.info("Title successfully created, id={}", title.getId()))
+          .onFailure(e -> log.error("Create title failed, id={}", title.getId(), e));
+      });
+  }
+
+  private Future<Void> populateTitle(Title title, PoLine poLine, Conn conn) {
+    if(Boolean.TRUE.equals(poLine.getIsPackage())) {
```

Comment:
Please format the file. Also minor thing but you can remove the `else` when `if` is terminated by `return`.